sup3r/preprocessing/dual_batch_handling.py

```diff
@@ -0,0 +1,297 @@
+        val_indices : list[dict]
+            List of dicts with handler_index and tuple_index. The tuple index
+            is used to get validation data observation with
+        data[tuple_index]
+        """
+
+        val_indices = []
+        for i, h in enumerate(self.data_handlers):
+            if h.hr_val_data is not None:
+                for _ in range(h.hr_val_data.shape[2]):
+                    spatial_slice = uniform_box_sampler(
+                        h.lr_val_data, self.lr_sample_shape[:2]
+                    )
+                    temporal_slice = uniform_time_sampler(
+                        h.lr_val_data, self.lr_sample_shape[2]
+                    )
+                    lr_index = tuple(
+                        [
+                            *spatial_slice,
+                            temporal_slice,
+                            np.arange(h.lr_val_data.shape[-1]),
+                        ]
+                    )
+                    hr_index = []
+                    for s in lr_index[:2]:
+                        hr_index.append(
+                            slice(
+                                s.start * self.s_enhance,
+                                s.stop * self.s_enhance,
+                            )
+                        )
+                    for s in lr_index[2:-1]:
+                        hr_index.append(
+                            slice(
+                                s.start * self.t_enhance,
+                                s.stop * self.t_enhance,
+                            )
+                        )
+                    hr_index.append(lr_index[-1])
+                    hr_index = tuple(hr_index)
+                    val_indices.append(
+                        {
+                            'handler_index': i,
+                            'hr_index': hr_index,
+                            'lr_index': lr_index,
+                        }
+                    )
+        return val_indices
+
+    @property
+    def shape(self):
+        """Shape of full validation dataset across all handlers
+
+        Returns
+        -------
+        shape : tuple
+            (spatial_1, spatial_2, temporal, features)
+            With temporal extent equal to the sum across all data handlers time
+            dimension
+        """
+        time_steps = 0
+        for h in self.data_handlers:
+            time_steps += h.hr_val_data.shape[2]
+        return (
+            self.data_handlers[0].hr_val_data.shape[0],
+            self.data_handlers[0].hr_val_data.shape[1],
+            time_steps,
+            self.data_handlers[0].hr_val_data.shape[3],
+        )
+
+    @property
+    def hr_sample_shape(self):
+        """Get sample shape for high_res data"""
+        return self.data_handlers[0].hr_dh.sample_shape
+
+    @property
+    def lr_sample_shape(self):
+        """Get sample shape for low_res data"""
+        return self.data_handlers[0].lr_dh.sample_shape
+
+    def __next__(self):
```

Comment:
This method looks clean and well organized, nice job (except for the line break style that i'm not a fan of :))